File: robot-client/src/main/java/com/informatica/cliente/logica/HuesoConHueso.java
package com.informatica.cliente.logica;

import lombok.Getter;

/**
 *
 * @author Vamaya
 */
public class HuesoConHueso  extends Hueso{
    //VARIABLES
    @Getter private final Hueso huesoUnido;
    
    //CONSTRUCTOR
    public HuesoConHueso(Hueso huesoUnido, int xFinal, int yFinal, int xInicial, int yInicial, int ancho) {
        super(xFinal, yFinal, xInicial, yInicial, ancho);
        this.huesoUnido = huesoUnido;
    }
    
    //METODOS SOBRE ESCRITOS
    @Override
    public void girarHueso(int alphaGrados){
        super.girarHueso(alphaGrados);
        int xn = this.getXFinalDinamica();
        int yn= this.getYFinalDinamica();
        huesoUnido.cambiarCoordenadasBase(xn, yn);
        
    }
    
    @Override
    public void cambiarCoordenadasBase(int xTrasladada, int yTrasladada){
        super.cambiarCoordenadasBase(xTrasladada, yTrasladada);
        int xnf = this.getXFinalDinamica();
        int ynf= this.getYFinalDinamica();
        huesoUnido.cambiarCoordenadasBase(xnf, ynf);
    }
    
    
    
}

File: my-app/src/logica/HuesoConHueso.ts
import { Hueso } from './Hueso';

export class HuesoConHueso extends Hueso {
    huesoUnido: Hueso;

    constructor(huesoUnido: Hueso, xFinal: number, yFinal: number, xInicial: number, yInicial: number, ancho: number) {
        super(xFinal, yFinal, xInicial, yInicial, ancho);
        this.huesoUnido = huesoUnido;
    }

    //METODOS SOBRE ESCRITOS
    girarHueso(alphaGrados: number){
        super.girarHueso(alphaGrados);
        let xn: number = this.xFinalDinamica;
        let yn: number = this.yFinalDinamica;
        this.huesoUnido.cambiarCoordenadasBase(xn, yn);

    }

    cambiarCoordenadasBase(xTrasladada: number, yTrasladada: number){
        super.cambiarCoordenadasBase(xTrasladada, yTrasladada);
        let xnf: number = this.xFinalDinamica;
        let ynf: number = this.yFinalDinamica;
        this.huesoUnido.cambiarCoordenadasBase(xnf, ynf);
    }

}
